src/ext/handlersToResp.c

```diff
@@ -380,73 +382,131 @@ static RdbRes toRespFunction(RdbParser *p, void *userData, RdbBulk func) {
 
 /*** Handling raw ***/
 
+static RdbRes toRespRawBeginModuleAux(RdbParser *p, void *userData, RdbBulk name, int encver, int when, size_t size) {
```

Comment:
Few lines below it can be seen assigned to `ctx->rawCtx.rawSize = size;`
I will modify it to `rawSize`

Reply:
but how do we know the rawSize when we begin a module aux? where do you take that info from?
i have a feeling (didn't bother to look), that it's always 0, and not used in this case...

Reply:
Please read the comment:
```
typedef struct RawContext {

    /* aggType - The parser encounters difficulty determining the size of
     * certain types during their parsing because they lack a preceding header
     * that would indicate their size. This creates a problem when implementing
     * callback handlers for executing RESP2 RESTORE commands on a live server,
     * which requires the type's size to be sent at the beginning. To resolve
     * this issue, the parser gathers all payload data for these types and only
     * provides it to callback handlers once it reaches the end of the type and
     * knows its total size. However, for types like strings, whose sizes are
     * already known at the beginning, the parser will not aggregate the entire
     * payload if it is large enough, but stream it to handleFrag callback. */
    enum {
        AGG_TYPE_UNINIT,
        AGG_TYPE_ENTIRE_DATA,
        AGG_TYPE_PARTIALLY
    } aggType;
```

Reply:
ok, i didn't remember the difference between this parser and the one i was involved with.
i.e. in the other parser the aggregation was on the receiver's side, so when starting a new payload the size wasn't known.

So here, it aggregates the entire thing before calling "Begin", and then it repeatedly calls the "Frag" callback? (i suppose that's necessary since each of them may be sitting in a separate allocation).

so this one doesn't have the optional size hint callback, and the "First" frag concept is internal to the callee, right?

thanks for clearing this up...

Reply:
Exactly